did_parser/src/parsed_did.rs

```diff
@@ -0,0 +1,84 @@
+use std::convert::TryFrom;
+use std::fmt::{Display, Formatter};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::DidUrl;
+use crate::{error::ParseError, utils::parse::parse_did_method_id, DidRange};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Did {
+    did: String,
```

Comment:
Maybe we could rename the `did` field to `full_did` or something like that? Might make things more intuitive as why there's a `did` field in the `Did` struct.

Reply:
I think it is quite obvious from the types.

Reply:
At least a comment then :D?

Reply:
I think adding a comment because of this is even worse, but will do as you wish.

Reply:
Why worse? A comment above the struct or just above the field would help everyone else who's looking at the code understand what's going on faster. 

For you and me it might be obvious because we discussed about this for some time, but newcomers would be saved from having to look at some other code to know exactly what the field is for. 

Although this is just my own opinion. @Patrik-Stas what do you think?

Reply:
IMO comment is not only highly unnecessary in this case as the purpose of the field is obvious from context but almost detrimental as one has read the code and the superfluous comment as well.

Reply:
As you wish, it's not a big deal.